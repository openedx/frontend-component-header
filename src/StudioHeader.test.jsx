/* eslint-disable react/prop-types */
import React, { useMemo } from 'react';
import { IntlProvider } from '@edx/frontend-platform/i18n';
import TestRenderer from 'react-test-renderer';
import { Link } from 'react-router-dom';
import { AppContext } from '@edx/frontend-platform/react';
import {
  ActionRow,
  Button,
  Dropdown,
} from '@edx/paragon';

import { StudioHeader } from './index';

const StudioHeaderComponent = ({ contextValue, appMenu = null, mainMenu = [] }) => (
  <IntlProvider locale="en" messages={{}}>
    <AppContext.Provider
      value={contextValue}
    >
      <StudioHeader appMenu={appMenu} mainMenu={mainMenu} />
    </AppContext.Provider>
  </IntlProvider>
);

const StudioHeaderContext = ({ actionRowContent = null }) => {
  const headerContextValue = useMemo(() => ({
    authenticatedUser: {
      userId: 'abc123',
      username: 'edX',
      roles: [],
      administrator: false,
    },
    config: {
      STUDIO_BASE_URL: process.env.STUDIO_BASE_URL,
      SITE_NAME: process.env.SITE_NAME,
      LOGIN_URL: process.env.LOGIN_URL,
      LOGOUT_URL: process.env.LOGOUT_URL,
      LOGO_URL: process.env.LOGO_URL,
    },
  }), []);
  return (
    <IntlProvider locale="en" messages={{}}>
      <AppContext.Provider
        value={headerContextValue}
      >
        <StudioHeader actionRowContent={actionRowContent} />
      </AppContext.Provider>
    </IntlProvider>
  );
};

describe('<StudioHeader />', () => {
  it('renders correctly', () => {
    const contextValue = {
      authenticatedUser: {
        userId: 'abc123',
        username: 'edX',
        roles: [],
        administrator: false,
      },
      config: {
        STUDIO_BASE_URL: process.env.STUDIO_BASE_URL,
        SITE_NAME: process.env.SITE_NAME,
        LOGIN_URL: process.env.LOGIN_URL,
        LOGOUT_URL: process.env.LOGOUT_URL,
        LOGO_URL: process.env.LOGO_URL,
      },
    };

    const component = <StudioHeaderComponent contextValue={contextValue} />;

    const wrapper = TestRenderer.create(component);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders correctly with optional action row content', () => {
    const actionRowContent = (
      <>
        <Dropdown>
          <Dropdown.Toggle variant="outline-primary" id="library-header-menu-dropdown">
            Settings
          </Dropdown.Toggle>
          <Dropdown.Menu>
            <Dropdown.Item as={Link} to="#">Dropdown Item 1</Dropdown.Item>
            <Dropdown.Item as={Link} to="#">Dropdown Item 2</Dropdown.Item>
            <Dropdown.Item as={Link} to="#">Dropdown Item 3</Dropdown.Item>
          </Dropdown.Menu>
        </Dropdown>
        <ActionRow.Spacer />
        <Button
          variant="tertiary"
          href="#"
          rel="noopener noreferrer"
          target="_blank"
          title="Help Button"
        >Help
        </Button>
      </>
    );

    const component = <StudioHeaderContext actionRowContent={actionRowContent} />;

    const wrapper = TestRenderer.create(component);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
