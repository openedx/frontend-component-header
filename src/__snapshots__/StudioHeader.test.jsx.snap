// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StudioHeader /> renders correctly 1`] = `
<header
  className="site-header-desktop"
>
  <a
    className="nav-skip sr-only sr-only-focusable"
    href="#main"
  >
    Skip to main content
  </a>
  <div
    className="container-fluid null"
  >
    <div
      className="nav-container position-relative d-flex align-items-center"
    >
      <img
        alt="edX"
        className="logo"
        src="https://edx-cdn.org/v3/default/logo.svg"
      />
      <div
        className="pgn__action-row"
      >
        <nav
          aria-label="Secondary"
          className="nav secondary-menu-container align-items-center ml-auto"
        >
          <div
            className="menu null"
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <button
              aria-expanded={false}
              aria-haspopup="menu"
              aria-label="Account menu for edX"
              className="menu-trigger btn btn-outline-primary d-inline-flex align-items-center pl-2 pr-3"
              onClick={[Function]}
            >
              <span
                className="avatar overflow-hidden d-inline-flex rounded-circle mr-2"
                style={
                  Object {
                    "height": "1.5em",
                    "width": "1.5em",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  focusable="false"
                  height="24px"
                  role="img"
                  style={
                    Object {
                      "height": "1.5em",
                      "width": "1.5em",
                    }
                  }
                  version="1.1"
                  viewBox="0 0 24 24"
                  width="24px"
                >
                  <path
                    d="M4.10255106,18.1351061 C4.7170266,16.0581859 8.01891846,14.4720277 12,14.4720277 C15.9810815,14.4720277 19.2829734,16.0581859 19.8974489,18.1351061 C21.215206,16.4412566 22,14.3122775 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,14.3122775 2.78479405,16.4412566 4.10255106,18.1351061 Z M12,24 C5.372583,24 0,18.627417 0,12 C0,5.372583 5.372583,0 12,0 C18.627417,0 24,5.372583 24,12 C24,18.627417 18.627417,24 12,24 Z M12,13 C9.790861,13 8,11.209139 8,9 C8,6.790861 9.790861,5 12,5 C14.209139,5 16,6.790861 16,9 C16,11.209139 14.209139,13 12,13 Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              edX
               
              <svg
                aria-hidden={true}
                focusable="false"
                height="16px"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16px"
              >
                <path
                  d="M7,4 L7,8 L11,8 L11,10 L5,10 L5,4 L7,4 Z"
                  fill="currentColor"
                  transform="translate(8.000000, 7.000000) rotate(-45.000000) translate(-8.000000, -7.000000) "
                />
              </svg>
            </button>
          </div>
        </nav>
      </div>
    </div>
  </div>
</header>
`;

exports[`<StudioHeader /> renders correctly with optional action row content 1`] = `
<header
  className="site-header-desktop"
>
  <a
    className="nav-skip sr-only sr-only-focusable"
    href="#main"
  >
    Skip to main content
  </a>
  <div
    className="container-fluid null"
  >
    <div
      className="nav-container position-relative d-flex align-items-center"
    >
      <img
        alt="edX"
        className="logo"
        src="https://edx-cdn.org/v3/default/logo.svg"
      />
      <div
        className="pgn__action-row"
      >
        <div
          className="dropdown"
          data-testid="dropdown"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-outline-primary"
            disabled={false}
            id="library-header-menu-dropdown"
            onClick={[Function]}
            type="button"
          >
            Settings
          </button>
        </div>
        <span
          className="pgn__action-row-spacer"
        />
        <a
          className="btn btn-tertiary"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          rel="noopener noreferrer"
          role="button"
          target="_blank"
          title="Help Button"
        >
          Help
        </a>
        <nav
          aria-label="Secondary"
          className="nav secondary-menu-container align-items-center ml-auto"
        >
          <div
            className="menu null"
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <button
              aria-expanded={false}
              aria-haspopup="menu"
              aria-label="Account menu for edX"
              className="menu-trigger btn btn-outline-primary d-inline-flex align-items-center pl-2 pr-3"
              onClick={[Function]}
            >
              <span
                className="avatar overflow-hidden d-inline-flex rounded-circle mr-2"
                style={
                  Object {
                    "height": "1.5em",
                    "width": "1.5em",
                  }
                }
              >
                <svg
                  aria-hidden={true}
                  focusable="false"
                  height="24px"
                  role="img"
                  style={
                    Object {
                      "height": "1.5em",
                      "width": "1.5em",
                    }
                  }
                  version="1.1"
                  viewBox="0 0 24 24"
                  width="24px"
                >
                  <path
                    d="M4.10255106,18.1351061 C4.7170266,16.0581859 8.01891846,14.4720277 12,14.4720277 C15.9810815,14.4720277 19.2829734,16.0581859 19.8974489,18.1351061 C21.215206,16.4412566 22,14.3122775 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,14.3122775 2.78479405,16.4412566 4.10255106,18.1351061 Z M12,24 C5.372583,24 0,18.627417 0,12 C0,5.372583 5.372583,0 12,0 C18.627417,0 24,5.372583 24,12 C24,18.627417 18.627417,24 12,24 Z M12,13 C9.790861,13 8,11.209139 8,9 C8,6.790861 9.790861,5 12,5 C14.209139,5 16,6.790861 16,9 C16,11.209139 14.209139,13 12,13 Z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              edX
               
              <svg
                aria-hidden={true}
                focusable="false"
                height="16px"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16px"
              >
                <path
                  d="M7,4 L7,8 L11,8 L11,10 L5,10 L5,4 L7,4 Z"
                  fill="currentColor"
                  transform="translate(8.000000, 7.000000) rotate(-45.000000) translate(-8.000000, -7.000000) "
                />
              </svg>
            </button>
          </div>
        </nav>
      </div>
    </div>
  </div>
</header>
`;
